server/routes/bookedtoday/arrivals/searchforexisting/changeDateOfBirthController.ts

```diff
@@ -4,19 +4,30 @@ import { State } from './state'
 export default class ChangeDateOfBirthController {
   public showChangeDateOfBirth(): RequestHandler {
     return async (req, res) => {
-      const data = State.searchDetails.get(req)
-      res.render('pages/bookedtoday/arrivals/searchForExistingRecord/changeDateOfBirth.njk', { data })
+      const data = req.flash('input')[0] || State.searchDetails.get(req)
+      res.render('pages/bookedtoday/arrivals/searchForExistingRecord/changeDateOfBirth.njk', {
+        data,
+        errors: req.flash('errors'),
+      })
     }
   }
 
   public changeDateOfBirth(): RequestHandler {
     return async (req, res) => {
       const { id } = req.params
+
+      if (req.errors) {
+        req.flash('input', req.body)
+        return res.redirect(`/prisoners/${id}/search-for-existing-record/change-date-of-birth`)
+      }
+
       const { day, month, year } = req.body
-      // TODO build and validate
-      State.searchDetails.update(req, res, { dateOfBirth: `${year}-${month}-${day}` })
 
-      res.redirect(`/prisoners/${id}/search-for-existing-record`)
+      State.searchDetails.update(req, res, {
+        dateOfBirth: `${year.padStart(4, '0')}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`,
```

Comment:
safe to build this now as validated - need to pad year or moment starts getting fancy and interpreting the fields in the wrong way - even with strict mode